crates/precompiles/src/tip20/rewards.rs

```diff
@@ -30,6 +31,13 @@ impl TIP20Token {
             return Err(TIP20Error::invalid_amount().into());
         }
 
+        // Only check access keys after Allegro Moderato hardfork
+        if self.storage.spec().is_allegro_moderato() {
+            // Check and update spending limits for access keys
+            let mut keychain = AccountKeychain::new();
+            keychain.authorize_transfer(msg_sender, token_address, call.amount)?;
+        }
```

Comment:
If this is related to transfer authorization, we could consider moving this to into `ensure_transfer_authorized` and passing in the call amount. Then we don't have to repeat this logic everywhere.